docs/sources/tasks/collect-logs-in-kubernetes.md

```diff
@@ -0,0 +1,323 @@
+  // Concatenate values __meta_kubernetes_pod_uid/__meta_kubernetes_pod_container_name.log
+  rule {
+    source_labels = ["__meta_kubernetes_pod_uid", "__meta_kubernetes_pod_container_name"]
+    action = "replace"
+    target_label = "__path__"
+    separator = "/"
+    replacement = "/var/log/pods/*$1/*.log"
+  }
+
+  // Label creation -  "container_runtime" field from "__meta_kubernetes_pod_container_id"
+  rule {
+    source_labels = ["__meta_kubernetes_pod_container_id"]
+    action = "replace"
+    target_label = "container_runtime"
+    regex = "^(\\S+):\\/\\/.+$"
+    replacement = "$1"
+  }
+}
+
+// loki.source.kubernetes tails logs from Kubernetes containers using the Kubernetes API.
+loki.source.kubernetes "pod_logs" {
+  targets    = discovery.relabel.pod_logs.output
+  forward_to = [loki.process.pod_logs.receiver]
+}
+
+// loki.process receives log entries from other Loki components, applies one or more processing stages,
+// and forwards the results to the list of receivers in the component’s arguments.
+loki.process "pod_logs" {
+  stage.static_labels {
+      values = {
+        cluster = "<CLUSTER_NAME>",
+      }
+  }
+
+  forward_to = [loki.write.<WRITE_COMPONENT_NAME>.receiver]
+}
+```
+
+Replace the following values:
+
+- _`<CLUSTER_NAME>`_: The label for this specific Kubernetes cluster, such as `production` or `us-east-1`.
+- _`<WRITE_COMPONENT_NAME>`_: The name of your `loki.write` component, such as `default`.
+
+### Kubernetes Cluster Events
+
+The following components are needed:
+
+1. [loki.source.kubernetes_events][]: Tails events from Kubernetes API
+1. [loki.process][]: Modify the logs before sending them to the next component
+1. [loki.write][]: Send logs to the Loki endpoint. You should have configured it in the [Configure logs delivery](#configure-logs-delivery) section
+
+Here is an example using those stages:
+
+```alloy
+// loki.source.kubernetes_events tails events from the Kubernetes API and converts them
+// into log lines to forward to other Loki components.
+loki.source.kubernetes_events "cluster_events" {
+  job_name   = "integrations/kubernetes/eventhandler"
+  log_format = "logfmt"
+  forward_to = [
+    loki.process.cluster_events.receiver,
+  ]
+}
+
+// loki.process receives log entries from other loki components, applies one or more processing stages,
+// and forwards the results to the list of receivers in the component’s arguments.
+loki.process "cluster_events" {
+  forward_to = [loki.write.<WRITE_COMPONENT_NAME>.receiver]
+
+  stage.static_labels {
+    values = {
+      cluster = "<CLUSTER_NAME>",
+    }
+  }
+
+  stage.labels {
+    values = {
+      kubernetes_cluster_events = "job",
+    }
+  }
+}
+```
+
+Replace the following values:
+
+- _`<CLUSTER_NAME>`_: The label for this specific Kubernetes cluster, such as `production` or `us-east-1`.
+- _`<WRITE_COMPONENT_NAME>`_: The name of your `loki.write` component, such as `default`.
+
```

Comment:
Would it be possible to add an example k8s configuration too, please? E.g. something similar to the example config here:
https://grafana.com/docs/alloy/latest/reference/components/otelcol.exporter.loadbalancing/#kubernetes-resolver

It could demonstrate that we need to store a positions file in a persistent way so that if Alloy is restarted, it can read it again.